osu.Game/Screens/Select/Carousel/CarouselGroup.cs

```diff
@@ -81,12 +83,9 @@ public override void Filter(FilterCriteria criteria)
         {
             base.Filter(criteria);
 
-            var children = new List<CarouselItem>(InternalChildren);
-
-            children.ForEach(c => c.Filter(criteria));
-            children.Sort((x, y) => x.CompareTo(criteria, y));
-
-            InternalChildren = children;
+            InternalChildren.ForEach(c => c.Filter(criteria));
+            // IEnumerable<T>.OrderBy() is used instead of List<T>.Sort() to ensure sorting stability
+            InternalChildren = InternalChildren.OrderBy(c => c, new CriteriaComparer(criteria)).ToList();
```

Comment:
Can be simplified via `Comparer<CarouselItem>.Create((x, y) => x.CompareTo(criteria, y))`

Reply:
Applied - this causes a slight semantics change since a `NullReferenceException` will be thrown on null instead of an `ArgumentNullException`, but if it's going to die on an exception it doesn't matter which one too much.